File: solutions/go/space-age/1/space_age.go
package space

type Planet string

const secondsPerYear = 60 * 60 * 24 * 365.25

func Age(seconds float64, planet Planet) float64 {
	var orbitalPeriod float64

	switch planet {
	case "Mercury":
		orbitalPeriod = 0.2408467
	case "Venus":
		orbitalPeriod = 0.61519726
	case "Earth":
		orbitalPeriod = 1.0
	case "Mars":
		orbitalPeriod = 1.8808158
	case "Jupiter":
		orbitalPeriod = 11.862615
	case "Saturn":
		orbitalPeriod = 29.447498
	case "Uranus":
		orbitalPeriod = 84.016846
	case "Neptune":
		orbitalPeriod = 164.79132
	default:
		return -1
	}

	return (seconds / secondsPerYear) / orbitalPeriod
}

File: solutions/typescript/space-age/3/space-age.ts
// enum Planet {
//   Mercury = 'mercury',
//   Venus = 'venus',
//   Earth = 'earth',
//   Mars = 'mars',
//   Jupiter = 'jupiter',
//   Saturn = 'saturn',
//   Uranus = 'uranus',
//   Neptune = 'neptune'
// }

enum OrbitalPeriods {
  mercury = 0.2408467,
  venus = 0.61519726,
  earth = 1,
  mars = 1.8808158,
  jupiter = 11.862615,
  saturn = 29.447498,
  uranus = 84.016846,
  neptune = 164.79132,
}

const SECONDS_PER_YEAR: number = 60 * 60 * 24 * 365.25

export function age(planet: keyof typeof OrbitalPeriods, seconds: number): number {
  const age = seconds / SECONDS_PER_YEAR / OrbitalPeriods[planet]

  return Number(age.toFixed(2))
}

